deploy/helm/sumologic/conf/setup/setup.sh

```diff
@@ -1,36 +1,89 @@
 #!/bin/bash
-cp /etc/terraform/{locals,main,providers,resources,variables}.tf /terraform
-cd /terraform
 
 # Fix URL to remove "v1" or "v1/"
 export SUMOLOGIC_BASE_URL=${SUMOLOGIC_BASE_URL%v1*}
-
 # Support proxy for terraform
 export HTTP_PROXY=${HTTP_PROXY:=""}
 export HTTPS_PROXY=${HTTPS_PROXY:=""}
 export NO_PROXY=${NO_PROXY:=""}
 
+function remaining_fields() {
+    local RESPONSE="$(curl -XGET -s \
+        -u "${SUMOLOGIC_ACCESSID}:${SUMOLOGIC_ACCESSKEY}" \
+        "${SUMOLOGIC_BASE_URL}"v1/fields/quota)"
+
+    echo "${RESPONSE}" | jq '.remaining'
+}
+
+# Check if we'd have at least 10 fields remaining after 8 additional fields
+# would be created for the collection
+function should_create_fields() {
+    local REMAINING=$(remaining_fields)
+    if [[ $(( REMAINING - 8 )) -ge 10 ]] ; then
+        echo 1
+    else
+        echo 0
+    fi
+}
+
+cp /etc/terraform/{locals,main,providers,resources,variables,fields}.tf /terraform/
+cd /terraform
+
 COLLECTOR_NAME="{{- if .Values.sumologic.collectorName }}{{ .Values.sumologic.collectorName }}{{- else}}{{ .Values.sumologic.clusterName }}{{- end}}"
 
 terraform init
 
-# Sumo Collector and HTTP sources
-terraform import sumologic_collector.collector "$COLLECTOR_NAME"
+# Sumo Logic fields
+readonly CREATE_FIELDS="$(should_create_fields)"
+if [[ "${CREATE_FIELDS}" -eq 0 ]]; then
+    echo "Couldn't automatically create fields\n"
+    echo "There's not enough fields which we could use for collection fields creation"
+    echo "Please free some of them and rerun the setup job"
+else
+    readonly FIELDS_RESPONSE="$(curl -XGET -s \
+        -u "${SUMOLOGIC_ACCESSID}:${SUMOLOGIC_ACCESSKEY}" \
+        "${SUMOLOGIC_BASE_URL}"v1/fields | jq '.data[]' )"
+
+    declare -ra FIELDS=("cluster" "container" "deployment" "host" "namespace" "node" "pod" "service")
```

Comment:
Could we move it to the `values.yaml`?

Reply:
Yes we could. We could also rewrite `fields.tf` to be a template looping over this array (declared in `values.yaml`)

Reply:
if we move fields to `values.yaml` I don't see any solution other than looping over it in `fields.tf`